### PR TITLE
py3k only support for edgar fetcher

### PR DESCRIPTION
**Pull request by aking1012**
I didn't test it at all on py2k and ripped out some of the tests, but it works for py3k.

### CODE REVIEW

File: pandas_datareader/tests/test_edgar.py [hunk 1]
```diff
@@ -24,49 +22,28 @@ def test_get_full_index(self):
...
-        except RemoteDataError as e:
-            raise nose.SkipTest(e)
-
     def test_6_digit_date(self):
-        # As of December 31, SEC has disabled ftp access to EDGAR.  Disabling tests until re-write
-        raise nose.SkipTest()
         try:
             ed = web.DataReader('daily', 'edgar-index', start='1998-05-18',
                                 end='1998-05-18')
+            print(type(ed))
```
Comment (bashtage) [inline pandas_datareader/tests/test_edgar.py]: Print should be removed

File: pandas_datareader/tests/test_edgar.py [hunk 2]
```diff
@@ -24,49 +22,28 @@ def test_get_full_index(self):
             raise nose.SkipTest(e)
 
     def test_get_nonzip_index_and_low_date(self):
-        # As of December 31, SEC has disabled ftp access to EDGAR.  Disabling tests until re-write
-        raise nose.SkipTest()
+        #Dailies now defaults to uncompressed, since gz is not always available
         try:
             ed = web.DataReader('daily', 'edgar-index', '1994-06-30',
                                 '1994-07-02')
+            print(type(ed))
```
Comment (bashtage) [inline pandas_datareader/tests/test_edgar.py]: Remove print

File: pandas_datareader/tests/test_edgar.py [hunk 3]
```diff
@@ -8,12 +8,10 @@
 
 
 class TestEdgarIndex(tm.TestCase):
-
     def test_get_full_index(self):
-        # As of December 31, SEC has disabled ftp access to EDGAR.  Disabling tests until re-write
-        raise nose.SkipTest()
         try:
             ed = web.DataReader('full', 'edgar-index')
+            print(ed)
```
Comment (bashtage) [inline pandas_datareader/tests/test_edgar.py]: Remove print

File: pandas_datareader/edgar.py [hunk 4]
```diff
@@ -182,68 +198,13 @@ def _sanitize_dates(self, start, end):
...
-                return True
-            else:
-                return False
-        except TypeError:
-            return False
+        #5 years and this isn't working...
+        timedelta = dt.timedelta(days=1825)
+        if end - start > timedelta:
+            end = start + timedelta
+            print("Be nice...")
```
Comment (bashtage) [inline pandas_datareader/edgar.py]: Remove prints

File: pandas_datareader/edgar.py [hunk 5]
```diff
@@ -50,124 +51,139 @@ class EdgarIndexReader(_BaseReader):
...
+                else:
+                    filedate = dt.datetime.strptime(idx_date, '%y%m%d')
+                    if filedate > _EDGAR_MAX_6_DIGIT_DATE:
+                        filedate = None
+        except AttributeError:
+            filedate = None   
+        return filedate
 
     def _sanitize_dates(self, start, end):
+        print("Sanitize gets called")
```
Comment (bashtage) [inline pandas_datareader/edgar.py]: Remove prints

File: pandas_datareader/edgar.py [hunk 6]
```diff
@@ -50,124 +51,139 @@ class EdgarIndexReader(_BaseReader):
...
-        return path
-
-    def read(self):
-        try:
-            self._sec_ftp_session = FTP(_SEC_FTP, timeout=self.timeout)
-            self._sec_ftp_session.login()
-        except EOFError:
-            raise RemoteDataError('FTP server has closed the connection.')
+    def _normalize_date_column(self, data):
+        print("In normalize date column")
```
Comment (bashtage) [inline pandas_datareader/edgar.py]: Remove print

File: pandas_datareader/edgar.py [hunk 7]
```diff
@@ -50,124 +51,139 @@ class EdgarIndexReader(_BaseReader):
...
+                    index_file.write(self._read_idx(self.url + instance + file_list_by_date[file_date]).read())
+        index_file.seek(0)
+        the_index = read_csv(index_file, delimiter='|', header=None,
+                         index_col=False, names=_COLUMNS,
+                         low_memory=False, dtype=_COLUMN_TYPES)
+        return the_index
+
+    def _read_zipfile(self, url):
+        print("In read zipfile")
+        zipf = BytesIO()
```
Comment (bashtage) [inline pandas_datareader/edgar.py]: Remove print

**Comment by aking1012** [comment]
I'll happily remove them, but I may need to add sleep commands in some places and retest.  Some things seemed to break sometimes with no prints.

**Comment by bashtage** [comment]
I don't think print can every be the correct solution.

**Comment by bashtage** [comment]
Also you need to pass flake8

e.g. in 

pandas-datareader, run

```bash
flake8 pandas_datareader
```

**Comment by aking1012** [comment]
Will do.  Do you mean the whole batch needs to pass or just the two files I touched?  flake8 raises a lot of style issues(276).  Around 50 are me.

**Comment by bashtage** [comment]
At a minimum you should make sure the files you touched pass flake8.

**Comment by bashtage** [comment]
From what I can see, all flake8 are in edgar or test_edgar

https://travis-ci.org/pydata/pandas-datareader/jobs/217471829

**Comment by aking1012** [comment]
Cool.  I'll fix the ones in edgar/test_edgar.  The others must be set to ignore.  gist: https://gist.github.com/aking1012/c1297814e5cc9eab957c7eeb6984156e

About half way through the edgar ones.

**Comment by femtotrader** [comment]
flake8 --ignore E501 pandas_datareader

**Comment by bashtage** [comment]
E501 is ignored.  See the .travis.yml file.

**Comment by aking1012** [comment]
Got it.  Thanks.

**Comment by jreback** [comment]
fyi you can create a setup.cfg like this: https://github.com/pandas-dev/pandas/blob/master/setup.cfg

to hold these defaults

**Comment by aking1012** [comment]
Fixed everything but the 501s.

**Comment by aking1012** [comment]
travis fails, but doesn't mention edgar at all now.  9 occurences in the logs, and it's all installation.

**Comment by aking1012** [comment]
@jreback - thanks for the config.  Not sure if this was a one and done or not though.  It could be worth the trouble if I get a hair to fix something else.

**Comment by jtkiley** [comment]
It looks like one of your commits replaces the all of the tests in `test_edgar.py` that you had previously edited with the contents of `edgar.py`. That's why there are no test mentions and coverage is low.

Also, it looks like we still support 2.7 (but thankfully not 2.6), so we may see some 2.7 compat issues once Travis runs tests on this code. Those should be much more reasonable to fix than 2.6 issues, though.

**Comment by aking1012** [comment]
Sorry.  Up arrow and a half edit on no sleep.  I'll fix it.

**Comment by gliptak** [comment]
@aking1012 Could you rebase? Thanks

**Comment by bashtage** [comment]
@aking1012 This seems to be broken now.  Any thought?

**Comment by aking1012** [comment]
I stopped working on it, because zero to something wasn't considered an improvement.  I haven't tried it out recently on my local copy.

**Comment by bashtage** [comment]
I think this is not easily fixable given the additional API changes in EDGAR.  I'm going to close now, but would welcome a working version.

**Pull request closed without merging**